packages/block-library/src/cover/edit/index.js

```diff
@@ -112,6 +112,10 @@ function CoverEdit( {
 	);
 	const mediaUrl = media?.source_url;
 
+	if ( mediaUrl && useFeaturedImage ) {
+		setCoverIsDark( mediaUrl, dimRatio, overlayColor.color );
+	}
```

Comment:
Can this logic also be moved inside the event handler? This will cause attribute updates on every render when a condition is true and trigger another re-render.

While it's okay to call `setState` during the render, I'm not sure if the same applies to the `setAttributes`.

Reply:
yeh, this is the one bit I wasn't sure on. Because the setting of featured image is outside the component we don't have access to the event handle, so maybe this is one case when an effect is better - will have a play.